### PR TITLE
Allow telemetry endpoint for JWT-limited requests

### PR DESCRIPTION
**Pull request by generall**
- **WIP: per-collection telemetry permissions**
- **upd telemetry**

Always answer something on `GET /telemetry` request. The difference now is that info inside telemetry will be filtered based on the JWT permissions.

Any form of global access can read everything

Collection level-access can only read related collections (collection stats + hardware usage), but can't read:

- requests data
- cluster info
- memory info

Collection level access with payload can't read collection info and hardware counters.

### CODE REVIEW

File: src/actix/api/service_api.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ fn telemetry(
     ActixAccess(access): ActixAccess,
 ) -> impl Future<Output = HttpResponse> {
     helpers::time(async move {
-        access.check_global_access(AccessRequirements::new())?;
+        // access.check_global_access(AccessRequirements::new())?;
```
Comment (timvisee) [inline src/actix/api/service_api.rs:42]:
```suggestion
```

**Review by timvisee — APPROVED** [review]
:raised_hands:

File: src/common/telemetry_ops/cluster_telemetry.rs [hunk 2]
```diff
@@ -72,11 +73,17 @@ pub struct ClusterTelemetry {
...
         detail: TelemetryDetail,
         dispatcher: &Dispatcher,
         settings: &Settings,
-    ) -> ClusterTelemetry {
-        ClusterTelemetry {
+    ) -> Option<ClusterTelemetry> {
+        let global_access = AccessRequirements::new().whole();
+        if access.check_global_access(global_access).is_err() {
+            return None;
+        }
```
Comment (coszio) [inline src/common/telemetry_ops/cluster_telemetry.rs:84]: nit
```suggestion
        access.check_global_access(AccessRequirements::new().whole()).ok()?
```

**Review by coszio — APPROVED** [review]
Didn't give it a run but it looks sane to me

**Pull request merged**